src/NuGet.Core/NuGet.ProjectManagement/Projects/FolderNuGetProject.cs

```diff
@@ -164,7 +164,7 @@ public string GetInstalledPath(PackageIdentity packageIdentity)
             return PackagePathResolver.GetInstalledPath(packageIdentity) ?? string.Empty;
         }
 
-        public async Task<bool> DeletePackage(PackageIdentity packageIdentity,
+        public Task<bool> DeletePackage(PackageIdentity packageIdentity,
```

Comment:
keep the method async (so the caller will get an error if he doesn't await) and then exceptions are going to be properly unrwapped